lucene/highlighter/src/java/org/apache/lucene/search/uhighlight/DefaultPassageFormatter.java

```diff
@@ -64,7 +64,7 @@ public String format(Passage[] passages, String content) {
     int pos = 0;
     for (Passage passage : passages) {
       // don't add ellipsis if its the first one, or if its connected.
-      if (passage.getStartOffset() > pos && pos > 0) {
+      if (passage.getStartOffset() != pos && pos > 0) {
```

Comment:
I ran this in a debugger to understand.  I think it'd be clearer if the code checks were done in the same order as the comment.  And the "first one" I think would now be clearer if checked via `sb.isEmpty` instead of `pos`, requiring reasoning about what `pos` could be in later iterations with out-of-order passages.

Reply:
```suggestion
      if (!sb.isEmpty() && passage.getStartOffset() != pos) {
```

Is this the expression you intended?

Reply:
Yes!

Reply:
I applied it to commit, [db47d0d](https://github.com/apache/lucene/pull/13832/commits/db47d0dd1ddc17cd5fc74ef93a968dcc1f0b31de)